tests/test_checksum.py

```diff
@@ -0,0 +1,21 @@
+import rasterio
+
+
+def test_checksum_band():
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        checksums = [src.checksum(i) for i in src.indexes]
+        assert checksums == [25420, 29131, 37860]
+
+
+def test_checksum_band_window():
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        window = ((0, src.height), (0, src.width))
```

Comment:
@sgillies How about a window smaller than the full extent of the data?  This proves that it doesn't fail when you give it a window, but doesn't prove that it calculates a different checksum for a smaller window...